htdocs/compta/paiement/class/paiement.class.php

```diff
@@ -1312,11 +1313,12 @@ public function fetch_thirdparty($force_thirdparty_id = 0)
 				$invoice = new Facture($this->db);
 				if ($invoice->fetch($billsarray[0]) > 0)
 				{
-					$force_thirdparty_id = $invoice->fk_soc;
+					$force_thirdparty_id = $invoice->socid;
 				}
 			}
 		}
-
-		return parent::fetch_thirdparty($force_thirdparty_id);
+        $soc=new Societe($this->db);
```

Comment:
Why not using the parent::fetch_thirdparty($force_thirdparty_id); ?

Reply:
because there is a bug on that function also. fetch_thirdparty returns 1 or 0 depending on the result of the operation and not the wanted thirdparty object.